[QA] Unapplied security updates for build container

## Description

The build container has unapplied security updates, causing builds to fail. We must update the build container in order to continue with 0.8.0 QA, specifically building 0.8.0~rc2 debs.

## Steps to Reproduce
1. Checkout `0.8.0-rc2` tag.
2. Run `make build-debs`

## Expected Behavior

Build succeeds without error.

## Actual Behavior

Build fails due to security upgrade check:

```
    ============================= test session starts ==============================
    platform linux2 -- Python 2.7.13, pytest-3.2.0, py-1.4.34, pluggy-0.4.0
    rootdir: /home/conor/freedomofpress/securedrop/molecule/builder, inifile:
    plugins: testinfra-1.12.0, xdist-1.18.2, forked-0.2
gw0 [177] / gw1 [177] / gw2 [177] / gw3 [177]
    scheduling tests via LoadScheduling
    .............................................................................................................F.....X.............................................................
    =================================== FAILURES ===================================
    _________ test_ensure_no_updates_avail[docker://trusty-sd-sec-update] __________
    [gw3] linux2 -- Python 2.7.13 /home/conor/.virtualenvs/sd/bin/python2
    
    host = <testinfra.host.Host object at 0x3a8fad76310>
    
        def test_ensure_no_updates_avail(host):
            """
                Test to make sure that there are no security-updates in the
                base builder container.
            """
        
            # Filter out all the security repos to their own file
            # without this change all the package updates appeared as if they were
            # coming from normal ubuntu update channel (since they get posted to both)
            host.run('egrep "^deb.*security" /etc/apt/sources.list > /tmp/sec.list')
        
            dist_upgrade_simulate = host.run('apt-get -s dist-upgrade '
                                             '-oDir::Etc::Sourcelist=/tmp/sec.list '
                                             '|grep "^Inst" |grep -i security')
        
            # If the grep was successful that means security package updates found
            # otherwise we get a non-zero exit code so no updates needed.
    >       assert dist_upgrade_simulate.rc != 0
    E       AssertionError: assert 0 != 0
    E        +  where 0 = CommandResult(command='apt-get -s dist-upgrade -oDir::Etc::Sourcelist=/tmp/sec...ntu3.3] (1:5.14-2ubuntu3.4 Ubuntu:14.04/trusty-security [all])\n', stderr=None).rc
    
    tests/test_security_updates.py:21: AssertionError
    ============== 1 failed, 175 passed, 1 xpassed in 111.01 seconds ===============
```

## Comments

We have lovely docs thanks to @msheiny: https://docs.securedrop.org/en/latest/development/dockerbuildmaint.html. Follow those, fix, submit PR.

--- a/molecule/builder/image_hash
+++ b/molecule/builder/image_hash
@@ -1,2 +1,2 @@
-# sha256 digest quay.io/freedomofpress/sd-docker-builder:2018_06_14
-69801e50e2ce899b4c4eba168dc8c10dffd89e0a35e19a406665c25346937f33
+# sha256 digest quay.io/freedomofpress/sd-docker-builder:2018_06_21
+b3a6a29e84407928511cb9f8ebdc9e62205abf02bfd9a6b9ccb02d52d814afe0

